packages/browser-wallet/src/shared/utils/token-helpers.ts

```diff
@@ -410,52 +405,42 @@ export async function getTokens(
     ids: string[],
     onError?: (error: string) => void
 ): Promise<GetTokensResult> {
-    const metadataPromise: Promise<[MetadataUrl[], Array<TokenMetadata | undefined>]> = (async () => {
-        let metadataUrls;
-        try {
-            metadataUrls = await getTokenUrl(client, ids, contractDetails);
-        } catch (e) {
-            onError?.(`Failed to get metadata urls on index: ${contractDetails.index}`);
-            return [[], []];
-        }
-        const metadata = await Promise.all(
-            metadataUrls.map((url, index) =>
-                getTokenMetadata(url).catch((error) => {
-                    onError?.(`id: "${ids[index]}": ${error.message}`);
-                    return Promise.resolve(undefined);
-                })
-            )
-        );
-
-        return [metadataUrls, metadata];
-    })();
-
-    const balancesPromise = getContractBalances(
-        client,
-        contractDetails.index,
-        contractDetails.subindex,
-        ids,
-        account,
-        onError
+    const tokenData: (TokenData | undefined)[] = await Promise.all(
+        ids.map(async (id, index) => {
+            let metadataUrl;
+            try {
+                [metadataUrl] = await getTokenUrl(client, [id], contractDetails);
```

Comment:
I think `getTokenUrl` is only ever invoked with a singleton list, so I would change the function to only support that.

Reply:
I just removed it, and used the helper from the SDK (did the same for getting balance)